script/watch-tests.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-console */
+
+const { fork } = require('child_process');
+const chokidar = require('chokidar');
+const util = require('util');
+const glob = util.promisify(require('glob'));
+const chalk = require('chalk');
+const _ = require('lodash');
+
+console.log(chalk.cyan('Performing initial set up'));
+
+const showErrors = process.argv.includes('showErrors');
+let watcher;
+let busy = false;
+let pendingTests = [];
+
+// all unit test files
+const allUnitTests = glob.sync('{src,test}/**/*.unit.spec.js?(x)');
+
+function runMochaTests(tests) {
+  // fork mocha into a separate process
+  const forked = fork('./script/mocha.js');
+
+  return new Promise((fulfill, reject) => {
+    // send the mocha process the tests to run
+    forked.send({
+      tests,
+      showErrors,
+    });
+    forked.on('message', ({ error, requiredFiles, unitTestsForSrc }) => {
+      if (error) {
+        // mocha runner returned an error
+        reject(error);
+      } else {
+        // mocha runner returend a list of files it required and a map of unit tests to source files
+        fulfill({ requiredFiles, unitTestsForSrc });
+      }
+      // kill mocha
+      forked.kill('SIGHUP');
+    });
+  });
+}
+
+// runs array of test files passed in or all tests if no value provided
+async function runUnitTests(unitTests = allUnitTests) {
+  if (busy) {
+    pendingTests = _.union(pendingTests, unitTests);
+    return;
+  }
+  console.log(chalk.cyan(`Running ${unitTests.length} unit test files`));
+
+  busy = true;
+  let unitTestsForSrc;
+  let requiredFiles;
+  try {
+    // eslint-disable-next-line no-undef
```

Comment:
Gonna be honest, I dunno. I didn't write it and I didn't want to spend the time refactoring it since it was supposed to be just a simple rename. I just couldn't commit it without fixing some of the linting errors...and ignoring others.